### PR TITLE
Add black as pre commit

### PR DESCRIPTION
**Pull request by volpatto**
In this PR, I'm adding `black` as pre-commit in order to solve #74.

Some remarks:

* A lot of modifications were introduced, mostly due to `black`. Thus, some errors are raised now, and I don't understand it, unless if I spend some time to study and understand `hookman` itself (I'm afraid that I don't have time fot this, please help me).

* The modifications that **I did**  are in [tox.in](https://github.com/volpatto/hookman/blob/f47a008a768ae142187897bcab86afa743684121/tox.ini), [.pre-commit-config.yaml](https://github.com/volpatto/hookman/blob/f47a008a768ae142187897bcab86afa743684121/.pre-commit-config.yaml) and [environment.devenv.yml](https://github.com/volpatto/hookman/blob/f47a008a768ae142187897bcab86afa743684121/environment.devenv.yml).

* Please note that I tried to use the most recent versions of `black` and `pre-commit` that are currently available in `conda-forge` channel.

* With black addition, we have a formatter and a checker (`flake8`) co-existing. This can be tricky. So I added some configuration to `flake8` in order to avoid conflicts. The configuration is based on [this](https://ljvmiranda921.github.io/notebook/2018/06/21/precommits-using-black-and-flake8/).

* At last, please help me to fix some errors. 3 tests are failing locally.

Thanks and I hope that you appreciate this PR.

### CODE REVIEW

**Comment by volpatto** [comment]
Well, the same three errors I had locally also occurred in the CI.

**Comment by volpatto** [comment]
> Well, the same three errors I had locally also occurred in the CI.

Ok, the errors are related to `pytest-datadir`. Unfortunately, I have no previous experiences with this plugin.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    -   id: black
+        args: [--safe, --quiet]
+        language_version: python3
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.3.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: debug-statements
+    -   id: flake8
```
Comment (williamjamir) [inline .pre-commit-config.yaml]: @nicoddemus What are your thoughts about `flake8`?

I think that we can drop it from pre-commit (recently added) and from the Tox also (already in place).

Comment (nicoddemus) [reply .pre-commit-config.yaml]: I agree we can drop it. 👍

**Comment by volpatto** [comment]
> Please add a `pyproject.toml` file with this contents:
> 
> ```toml
> [tool.black]
> line-length = 100
> ```
> 
> The default line length (88) is too short for us.



> Please add a `pyproject.toml` file with this contents:
> 
> ```toml
> [tool.black]
> line-length = 100
> ```
> 
> The default line length (88) is too short for us.

Sure, I will do that.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (volpatto) [reply .pre-commit-config.yaml]: Okay, I will remove `flake8` then. The formatter is enough, in my opinion.

**Comment by williamjamir** [comment]
> At last, please help me to fix some errors. 3 tests are failing locally.

To solve the issue, you need to re-run the test (locally) with the `--force-regen` parameter in order to re-generate the expected files ;)

The appveyor for python 3.7 is currently failing, so don't need to worry about that =/

> Thanks and I hope that you appreciate this PR.

Sure I do 😀

**Comment by volpatto** [comment]
@williamjamir @nicoddemus out of curiosity (sorry if this is a dumb question): why `tox` is not included in `environment.devenv.yml`?

**Comment by nicoddemus** [comment]
It is because `tox` is used only in virtual environments, not with conda

**Comment by volpatto** [comment]
blac

> It is because `tox` is used only in virtual environments, not with conda

Ah! Okay, thanks!

**Comment by volpatto** [comment]
> Please add a `pyproject.toml` file with this contents:
> 
> ```toml
> [tool.black]
> line-length = 100
> ```
> 
> The default line length (88) is too short for us.

Done.

**Comment by nicoddemus** [comment]
@volpatto did you run black again after changing `line-length`?

**Comment by volpatto** [comment]
> @volpatto did you run black again after changing `line-length`?

Yeah, it reduced the number of new inclusions in the diff.

**Pull request merged**